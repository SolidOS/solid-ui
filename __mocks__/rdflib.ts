export const Util =  {
  mediaTypeClass: sym
}

export function sym () {
  return {
    dir: sym,
    uri: 'uri',
    value: '',
    elements: [],
    doc: () => {
      return {
        equals: () => {},
        value: ''
      }
    },
    sameTerm: () => false
  }  
}  

export const uri = {
  join: () => {}
}
export function st() {
}

export function literal() {
}
export function graph() {
  return {
    any: () => {
      return sym()
    },
    each: () => {
      return []
    },
    match: () => {
      return []
    },
    query: () => {
      return []
    },
    fetcher: {
      load: () => {},
      nowOrWhenFetched: () => Promise.resolve()
    },
    findTypeURIs: () => {
      return []
    },
    findSuperClassesNT: () => {
      return []
    },
    bottomTypeURIs: () => {
      return []
    },
    statementsMatching: () => {
      return []
    },
    sym
  }
}
export function fetcher() {
  return {}
}
export class Fetcher {
  requested: any
  constructor() {
    this.requested = {}
  }
  load () {
  }
}
export function namedNode() {
  return {}
}
export function NamedNode() {
}
export function Namespace() {
  return () => {}
}
export class UpdateManager {
  editable() {}
  update() {}
}
