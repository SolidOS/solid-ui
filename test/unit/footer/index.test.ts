import { silenceDebugMessages } from '../helpers/debugger'
import {
  initFooter,
  rebuildFooter,
  createControllerInfoBlock
} from '../../../src/footer'
import { NamedNode } from 'rdflib'
// @ts-ignore
import { solidLogicSingleton } from 'solid-logic'

const store = solidLogicSingleton.store

silenceDebugMessages()

describe('footer', () => {
  it('is exposed on public API', async () => {
    const options = { solidProjectUrl: 'https://solid.com/', solidProjectName: 'Solid Project' }
    expect(await initFooter(store, options)).toMatchSnapshot()
  })
})
describe('rebuildFooter', () => {
  it('creates a link', () => {
    const footer = document.createElement('div')
    const pod = new NamedNode('https://test.com')
    const podOwner = new NamedNode('https://test2.com')
    const options = { solidProjectUrl: 'https://solid.com/', solidProjectName: 'Solid Project' }

    expect(rebuildFooter(footer, store, pod, podOwner, options)).toMatchSnapshot()
  })
  it('does NOT creates a link in the footer', () => {
    const footer = document.createElement('div')
    const pod = new NamedNode('https://test.com')
    const podOwner = new NamedNode('https://test.com')
    expect(rebuildFooter(footer, store, pod, podOwner)).toMatchSnapshot()
  })
})
describe('createControllerInfoBlock', () => {
  it('runs createControllerInfoBlock', async () => {
    const pod = new NamedNode('https://test.com')
    const podOwner = new NamedNode('https://test.com')
    const user = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    expect(createControllerInfoBlock(store, user, pod, podOwner)).toMatchSnapshot()
  })
  it('makes use of options', async () => {
    const pod = new NamedNode('https://test.com')
    const podOwner = new NamedNode('https://test.com')
    const user = new NamedNode('https://sharonstrats.inrupt.net/profile/card#me')
    const options = { solidProjectUrl: 'https://solid.com/', solidProjectName: 'Solid Project' }
    expect(createControllerInfoBlock(store, user, pod, podOwner, options)).toMatchSnapshot()
  })
})
