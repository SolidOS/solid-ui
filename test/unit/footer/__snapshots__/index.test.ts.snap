// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`createControllerInfoBlock makes use of options 1`] = `
<div
  style="border-top: solid 1px $divider-color; font-size: 0.9em; padding: 0.5em 1.5em;"
>
  <span />
  <a
    href="https://test.com"
  />
  <span />
  <a
    href="https://test.com"
  />
  <span />
  <a
    href="https://solid.com/"
  />
  <span />
</div>
`;

exports[`createControllerInfoBlock runs createControllerInfoBlock 1`] = `
<div
  style="border-top: solid 1px $divider-color; font-size: 0.9em; padding: 0.5em 1.5em;"
>
  <span />
  <a
    href="https://test.com"
  />
  <span />
  <a
    href="https://test.com"
  />
  <span />
  <a
    href="https://solidproject.org"
  />
  <span />
</div>
`;

exports[`footer is exposed on public API 1`] = `undefined`;

exports[`rebuildFooter creates a link 1`] = `[Function]`;

exports[`rebuildFooter does NOT creates a link in the footer 1`] = `[Function]`;
